File: algorithms/binary-search/Go/BinarySearch.go
package main

import "fmt"

func binarySearch(array []int, search int) int {
	low := 0
	high := len(array) - 1
	for low <= high {
		mid := (low + high) / 2
		if array[mid] == search {
			return mid
		}
		if array[mid] < search {
			low = mid + 1
		} else {
			high = mid - 1
		}
	}
	return -1
}

func main() {
	array := []int{1, 2, 3, 4, 5, 6, 7, 8, 9, 10}

	tests := []struct {
		search int
		wanted int
	}{
		{1, 0},
		{5, 4},
		{42, -1},
		{10, 9},
	}
	for _, tt := range tests {
		result := binarySearch(array, tt.search)
		if result != tt.wanted {
			fmt.Printf("got: %v, want: %v\n", result, tt.wanted)
		}
	}

}

File: algorithms/binary-search/PHP/BinarySearch.php
<?php

function binarySearch(array $array, int $search): int
{
    $low = 0;
    $high = count($array) - 1;
    while ($low <= $high) {
        $mid = intval(($low + $high) / 2);
        if ($array[$mid] == $search) {
            return $mid;
        }
        if ($array[$mid] < $search) {
            $low = $mid + 1;
        } else {
            $high = $mid - 1;
        }
    }
    return -1;
}

$array = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10];

$tests = [
    1 => 0,
    5 => 4,
    42 => -1,
    10 => 9,
];
foreach ($tests as $search => $wanted) {
    $result = binarySearch($array, $search);
    if ($result !== $wanted) {
        printf("got: %d, want: %d\n", $result, $wanted);
    }
}
